### DISCUSSION TITLE
How to deploy without iam role?

### DISCUSSION THREAD

**Discussion by le2sky** [serverless/serverless#11073] [Q&A] [answered]
Hello, Serverless people! i am a student developer in korea.
Recently, i am implementing the batch processing with AWS lambda, sqs, and serverless-framework for caching data at ElasticCache. 
I requested the aws iam permission to the aws administrator with the least privilege. however, i dont think it's not good for iam account to have the iam role. Is there a way to deploy a function without an iam role? plz help me

the way i thought without iam role:
1. Do not use  serverless framework and  Deploy directly to aws lambda with docker image or zip without any permission.

minimum privileges referring to the following issues:
[#1439 Narrowing the Serverless IAM Deployment Policy](https://github.com/serverless/serverless/issues/1439)

**Comment by pgrzesik** [comment, marked as answer]
Hello @le2sky - if I understand correctly, you would like to use permissions for a user that does not have policy that allows IAM Role creation, correct? Unfortunatelly, the IAM Role creation is needed as we need to create IAM role that will be attached to the Lambda functions that you deploy. You can bypass that by providing a role that already exists via `provider.iam.role` - in such situation, the Framework will use that role for Lambda functions instead of creating a new one.

**Reply by le2sky** [reply]
You got the gist of the question exactly and it will be helpful to me. Thanks for your kind reply.👍
